pkgs/applications/science/biology/bwa-mem2/default.nix

```
@@ -0,0 +1,47 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, zlib
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "bwa-mem2";
+  version = "unstable-2023-03-18";
+
+  src = fetchFromGitHub {
+    owner = "bwa-mem2";
+    repo = "bwa-mem2";
+    rev = "cf4306a47dac35e7e79a9e75398a35f33900cfd0";
+    fetchSubmodules = true;
+    hash = "sha256-1AYSn7nBrDwbX7oSrdEoa1d3t6xzwKnA0S87Y/XeXJg=";
+  };
+
+  buildInputs = [ zlib ];
+
+  # see https://github.com/bwa-mem2/bwa-mem2/issues/93
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    sed -i 's/memset_s/memset8_s/g' ext/safestringlib/include/safe_mem_lib.h
+    sed -i 's/memset_s/memset8_s/g' ext/safestringlib/safeclib/memset16_s.c
+    sed -i 's/memset_s/memset8_s/g' ext/safestringlib/safeclib/memset32_s.c
+    sed -i 's/memset_s/memset8_s/g' ext/safestringlib/safeclib/memset_s.c
+    sed -i 's/memset_s/memset8_s/g' ext/safestringlib/safeclib/wmemset_s.c
+  '';
+
```

```suggestion

  buildFlags = [
    (if stdenv.hostPlatform.sse4_2Support then "arch=sse42"
    else if stdenv.hostPlatform.avxSupport then "arch=avx"
    else if stdenv.hostPlatform.avx2Support then "arch=avx2"
    else if stdenv.hostPlatform.avx512Support then "arch=avx512"
    else "arch=sse41")
  ];
```
Some binaries are generated depending on the optimization level, but we only need one. 
Using the build flag, we get the appropriate optimization.
ref) https://github.com/bwa-mem2/bwa-mem2/blob/cf4306a47dac35e7e79a9e75398a35f33900cfd0/Makefile#L54